### PR TITLE
Roll-back snippet indentation

### PR DESCRIPTION
**Pull request by ennru**
This rolls back the snippet indentation logic introduced by #201. But has special handling for `import` lines, so that they become aligned with the left-most code in other snippets with the same label.

Relates to:
https://github.com/akka/akka/issues/25564
https://github.com/lightbend/paradox/pull/242

### CODE REVIEW

File: core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala [hunk 1]
```diff
@@ -28,16 +26,40 @@ object Snippet {
   class SnippetException(message: String) extends RuntimeException(message)
 
   def apply(file: File, labels: Seq[String]): (String, String) = {
-    (extract(file, labels), language(file))
+    val lines = Source.fromFile(file)("UTF-8").getLines.toSeq
```
Comment (ktoso) [inline core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala]: this leaks file descriptor doesnt it?
gotta close the source

File: core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala [hunk 2]
```diff
@@ -28,16 +26,40 @@ object Snippet {
...
+    (extract(file, lines, labels), language(file))
+  }
+
+  def extract(file: File, lines: Seq[String], labels: Seq[String]): String = {
```
Comment (ktoso) [inline core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala]: extractSnippet right?

Comment (ktoso) [reply core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala]: (just a nitpick on naming)

File: core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala (hunk 1 above)
Comment (ennru) [reply core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala]: Good point, the lines used to be read multiple times like this...

Comment (ktoso) [reply core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala]: There's a few spots below as well, please skim for the leak :)

File: core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala [hunk 3]
```diff
@@ -28,16 +26,40 @@ object Snippet {
...
+            Math.min(in, idx)
+          else in
+        case (in, _) => in
+      }
+    snippetLines.map(ln => dropIndent(minIndent, ln)).mkString("\n")
   }
 
   def extractLabelRange(file: File, label: String): Option[(Int, Int)] = {
-    val lineNumbers = extractState(file, label).lines.map(_._1)
+    val lines = Source.fromFile(file)("UTF-8").getLines.toSeq
```
Comment (ktoso) [inline core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala]: here too, watch out to close the source

File: tests/src/test/scala/com/lightbend/paradox/markdown/SnipDirectiveSpec.scala [hunk 4]
```diff
@@ -89,7 +89,7 @@ class SnipDirectiveSpec extends MarkdownBaseSpec {
       |<pre class="prettyprint">
       |<code class="language-scala">
       |case object Dent
-      |case object DoubleDent</code>
+      |  case object DoubleDent</code>
```
Comment (ktoso) [inline tests/src/test/scala/com/lightbend/paradox/markdown/SnipDirectiveSpec.scala]: this one seems suprising?

File: core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala (hunk 2 above)
Comment (ennru) [reply core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala]: I guess `Snippet.extract` is OK.

File: tests/src/test/scala/com/lightbend/paradox/markdown/SnipDirectiveSpec.scala (hunk 4 above)
Comment (ennru) [reply tests/src/test/scala/com/lightbend/paradox/markdown/SnipDirectiveSpec.scala]: That's what Akka asked for...
The example is for how Alpakka wanted it to be.

File: tests/src/test/scala/com/lightbend/paradox/markdown/SnippetIndentationTest.scala [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        |    out.println(i);
+        |}""".stripMargin
+    )
+  }
+
+  def extractToString(inString: String, label: String, indentationPerSnippet: Boolean = true): String = {
+    val in = inString.split("\n").toList
+    Snippet.extract(new File(""), in, Seq(label))
+  }
+}
```
Comment (ktoso) [inline tests/src/test/scala/com/lightbend/paradox/markdown/SnippetIndentationTest.scala]: nice test additions

**Pull request merged**